src/nodes/helper.ts

```diff
@@ -98,13 +92,14 @@ function extendEvent(event: IKalenderEvent, config: IcalEventsConfig, kalenderEv
 }
 
 export async function getICal(node: IcalNode, RED,n:any) {
-    const kalenderEvents = new KalenderEvents()
+    const kalenderEvents = node.ke ? node.ke : new KalenderEvents();
```

Comment:
Nice work, thanks a lot for digging into the issue 👍
The only thing I'm not sure about is this. I'm not 100% sure if this doesn't cause anyside affects, when my lib kalender-events is a singleton here 😉
But let's have a try

Reply:
Heh, hopefully your kalender-events is solid!  If we have troubles with that object being reused then I'll just have to dig into it and figure out why.  :)  So far I haven't had any issues.